[Bug]: The plugin encountered an error, and failed to respond to the │ plugin.(*GRPCProvider).ApplyResourceChange call. Plugin did not respond

### Terraform Core Version

1.8.5

### AWS Provider Version

5.54.0

### Affected Resource(s)

* aws_iot_thing

### Expected Behavior

IOT thing Successful creation

### Actual Behavior

```console
╷
│ Error: Plugin did not respond
│ 
│   with aws_iot_thing.this["thing2"],
│   on main.tf line 2, in resource "aws_iot_thing" "this":
│    2: resource "aws_iot_thing" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-aws_v5.54.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 60 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/iot.expandThingGroupProperties(0xc0033e9ec0)
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:277 +0x17e
github.com/hashicorp/terraform-provider-aws/internal/service/iot.resourceThingGroupCreate({0x15778028, 0xc0033e9b00}, 0xc0033b6780, {0x137ca7e0, 0xc0025745b0})
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:146 +0x27e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc0033b6780?, {0x137ca7e0?, 0xc0025745b0})
        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x15778028?, {0x15778028?, 0xc0033a77d0?}, 0xd?, {0x137ca7e0?, 0xc0025745b0?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0010bf6c0, {0x15778028, 0xc0033a77d0}, 0xc0031f6f70, 0xc0033b6600, {0x137ca7e0, 0xc0025745b0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc002821d70, {0x15778028?, 0xc0033a7710?}, 0xc0033dc140)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc000b78770, {0x15778028?, 0xc0033a7440?}, 0xc0033dc140)
        github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000672f00, {0x15778028?, 0xc0033a6c60?}, 0xc0008e2850)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x13594e60, 0xc000672f00}, {0x15778028, 0xc0033a6c60}, 0xc0033b6280, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001066000, {0x15778028, 0xc0033a6bd0}, {0x157c9300, 0xc00144d500}, 0xc0033a5e60, 0xc00264dad0, 0x1e864378, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc001066000, {0x157c9300, 0xc00144d500}, 0xc0033a5e60)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 27
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.54.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Relevant Error/Panic Output Snippet

```console
╷
│ Error: Plugin did not respond
│ 
│   with aws_iot_thing.this["thing2"],
│   on main.tf line 2, in resource "aws_iot_thing" "this":
│    2: resource "aws_iot_thing" "this" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵


Stack trace from the terraform-provider-aws_v5.54.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 60 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/iot.expandThingGroupProperties(0xc0033e9ec0)
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:277 +0x17e
github.com/hashicorp/terraform-provider-aws/internal/service/iot.resourceThingGroupCreate({0x15778028, 0xc0033e9b00}, 0xc0033b6780, {0x137ca7e0, 0xc0025745b0})
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:146 +0x27e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc0033b6780?, {0x137ca7e0?, 0xc0025745b0})
        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x15778028?, {0x15778028?, 0xc0033a77d0?}, 0xd?, {0x137ca7e0?, 0xc0025745b0?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0010bf6c0, {0x15778028, 0xc0033a77d0}, 0xc0031f6f70, 0xc0033b6600, {0x137ca7e0, 0xc0025745b0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc002821d70, {0x15778028?, 0xc0033a7710?}, 0xc0033dc140)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc000b78770, {0x15778028?, 0xc0033a7440?}, 0xc0033dc140)
        github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000672f00, {0x15778028?, 0xc0033a6c60?}, 0xc0008e2850)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x13594e60, 0xc000672f00}, {0x15778028, 0xc0033a6c60}, 0xc0033b6280, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001066000, {0x15778028, 0xc0033a6bd0}, {0x157c9300, 0xc00144d500}, 0xc0033a5e60, 0xc00264dad0, 0x1e864378, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc001066000, {0x157c9300, 0xc00144d500}, 0xc0033a5e60)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 27
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.54.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

```hcl
terraform { 
  required_providers { 
    aws = { 
      source = "hashicorp/aws" 
      version = "~> 5.0" 
    } 
  } 
}

provider "aws" {
  region = "us-west-2"
}
```

### Steps to Reproduce

```hcl
resource "aws_iot_thing" "example" {
  name = "example"

  attributes = {
    First = "examplevalue"
  }
}
```

### Debug Output

```console
Stack trace from the terraform-provider-aws_v5.54.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 42 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/iot.expandThingGroupProperties(0xc002a8fa10)
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:277 +0x17e
github.com/hashicorp/terraform-provider-aws/internal/service/iot.resourceThingGroupCreate({0x15778028, 0xc002a8f650}, 0xc0009fed80, {0x137ca7e0, 0xc0023341a0})
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:146 +0x27e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc0009fed80?, {0x137ca7e0?, 0xc0023341a0})
        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x15778028?, {0x15778028?, 0xc0026cd290?}, 0xd?, {0x137ca7e0?, 0xc0023341a0?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0012bfdc0, {0x15778028, 0xc0026cd290}, 0xc00268e8f0, 0xc0009fec00, {0x137ca7e0, 0xc0023341a0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00403cc78, {0x15778028?, 0xc0026cd1d0?}, 0xc0026a84b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001657810, {0x15778028?, 0xc0026ccf00?}, 0xc0026a84b0)
        github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000210500, {0x15778028?, 0xc0026cc720?}, 0xc000250540)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x13594e60, 0xc000210500}, {0x15778028, 0xc0026cc720}, 0xc0009fe680, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0014a0200, {0x15778028, 0xc0026cc690}, {0x157c9300, 0xc0005d2000}, 0xc0026cad80, 0xc0048d7ad0, 0x1e864378, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc0014a0200, {0x157c9300, 0xc0005d2000}, 0xc0026cad80)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.54.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Panic Output

```console
Stack trace from the terraform-provider-aws_v5.54.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 42 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/iot.expandThingGroupProperties(0xc002a8fa10)
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:277 +0x17e
github.com/hashicorp/terraform-provider-aws/internal/service/iot.resourceThingGroupCreate({0x15778028, 0xc002a8f650}, 0xc0009fed80, {0x137ca7e0, 0xc0023341a0})
        github.com/hashicorp/terraform-provider-aws/internal/service/iot/thing_group.go:146 +0x27e
github.com/hashicorp/terraform-provider-aws/internal/provider.New.(*wrappedResource).Create.interceptedHandler[...].func8(0xc0009fed80?, {0x137ca7e0?, 0xc0023341a0})
        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:113 +0x283
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x15778028?, {0x15778028?, 0xc0026cd290?}, 0xd?, {0x137ca7e0?, 0xc0023341a0?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:801 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0012bfdc0, {0x15778028, 0xc0026cd290}, 0xc00268e8f0, 0xc0009fec00, {0x137ca7e0, 0xc0023341a0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00403cc78, {0x15778028?, 0xc0026cd1d0?}, 0xc0026a84b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001657810, {0x15778028?, 0xc0026ccf00?}, 0xc0026a84b0)
        github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000210500, {0x15778028?, 0xc0026cc720?}, 0xc000250540)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x13594e60, 0xc000210500}, {0x15778028, 0xc0026cc720}, 0xc0009fe680, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0014a0200, {0x15778028, 0xc0026cc690}, {0x157c9300, 0xc0005d2000}, 0xc0026cad80, 0xc0048d7ad0, 0x1e864378, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc0014a0200, {0x157c9300, 0xc0005d2000}, 0xc0026cad80)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0xe8b
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 55
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x125

Error: The terraform-provider-aws_v5.54.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

None

--- a/.changelog/38776.txt
+++ b/.changelog/38776.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_iot_thing_group: Fix crash when empty `attribute_payload` block is configured
+```

--- a/internal/service/iot/thing_group.go
+++ b/internal/service/iot/thing_group.go
@@ -291,7 +291,7 @@ func expandThingGroupProperties(tfMap map[string]interface{}) *awstypes.ThingGro
 
 	apiObject := &awstypes.ThingGroupProperties{}
 
-	if v, ok := tfMap["attribute_payload"].([]interface{}); ok && len(v) > 0 {
+	if v, ok := tfMap["attribute_payload"].([]interface{}); ok && len(v) > 0 && v[0] != nil {
 		apiObject.AttributePayload = expandAttributePayload(v[0].(map[string]interface{}))
 	}
 

